docs/src/model_setup/buoyancy_and_equation_of_state.md

```diff
@@ -2,14 +2,15 @@
 
 The buoyancy option selects how buoyancy is treated. There are currently three options:
 
-1. No buoyancy (and no gravity).
+1. No buoyancy (and no gravity). This is the default setting.
 2. Evolve buoyancy as a tracer.
 3. _Seawater buoyancy_: evolve temperature ``T`` and salinity ``S`` as tracers with a value for the gravitational
-   acceleration ``g`` and an equation of state of your choosing. This is the default setting.
+   acceleration ``g`` and an equation of state of your choosing.
 
 ## No buoyancy
 
-To turn off buoyancy (and gravity) simply pass `buoyancy = nothing` to the model constructor.
+To turn off buoyancy (and gravity) you can simply create a model without explocitly passing the
+`buoyancy` flag:
```

Comment:
```suggestion
`buoyancy` flag since, by default, `buoyancy = nothing`.
```

Is this the default for *all* models or only for the `NonhydrostaticModel`?

Reply:
Just for `NonhydrostaticModel`. But that whole section is (I believe at least) focused on `NonhydrostaticModel`, no? I shuld say, I have no idea how buoyancy is treated or how the options are for the other models since I've never used them. So if this discussion needs to include other models (which even before this PR it doesn't) then I might need to assistance writing this part.

Reply:
The section implies that this is the default for all. But,
https://github.com/CliMA/Oceananigans.jl/blob/a1cd98817ba84233dd3a146813e3574ebc01ac9f/src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl#L87-L104

Reply:
We need to remedy this.

Makes me wonder if we should have just left the defaults as they were. All this work just to change the defaults is it really worth it? (This is a rhetorical question.)

Reply:
Probably we need to change the defaults for `HydrostaticFreeSurfaceModel`.

@glwagner?

Reply:
I don't think it's a problem to have to have the defaults be different for different models. We just need to make that explicit in this part of the docs. I think the best thing regardless of this choice is to have at least some part of the `model_setup` docs to be model-specific, which isn't the case at the moment.

Reply:
Sure. I agree, there is no problem for different defaults but the docs shouldn't imply otherwise.

Reply:
`Model Setup > Overview` describes all three models: https://clima.github.io/OceananigansDocumentation/stable/model_setup/overview/

There are only a few properties that are specific to one model or the other, because we've written the code so that both models use the same types for physics specification. The hydrostatic model has a `free_surface` property that isn't currently documented, and the hydrostatic model allows different schemes for `momentum_advection` and `tracer_advection` (though it looks like there's no section for advection in model setup right now anyways). @tomchor are you suggesting that some part of the model setup section needs to be model-specific in particular?

I agree its ok if the defaults are different, though I'm worried that this is surprising. This is of course why we used T, S as default tracers for the nonhydrostatic model (because eventually we anticipated that Oceananigans would be used primarily for large scale ocean modeling). Let's think about this a bit...

Reply:
I don't think it would be surprising to users that the default is different if it's well-documented. The issue is that it wasn't clear for me that this section of the docs (`model_setup/buoyancy`) was supposed to be for all models. In the previous version no other model was even mentioned in that section, which made me assume that section was specific to nonhydrostatic models. 

I still believe `nothing` is a better default for the hydrostatic model though (saves memory, more conservative https://github.com/CliMA/Oceananigans.jl/issues/1984; although I might be biased since I've invested several hours into this PR already), so I'll modify it to include at least mention of all models. Let's see how that looks before merging.

Reply:
The main issue is that hydrostatic simulations with no stratification / active tracers are quite rare so `nothing` as a default isn't really a convenience. Currently most models use `BuoyancyTracer`. But that might not be a good default, either. I'm ok with either `nothing`, or no default.

Reply:
Ah, damn, sorry, it was a typo that I fixed. I meant to say that I believe `nothing` is a better default for `NonhydrostaticModel`. But I agree that _hydrostatic_ simulations without any buoyancy treatment is indeed very uncommon. I think all tests pass now, and I've re-written the buoyancy part of model set-up in a way that (I think) is much more comprehensive. This PR doesn't change anything for `HydrostaticModel`.